components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java

```diff
@@ -166,6 +166,18 @@ default Resource replaceResource(String resourceTypeName, Resource resource)
      */
     Resource getResource(String resourceTypeName, String resourceName) throws ConfigurationManagementException;
 
+    /**
+     * This API is used to retrieve the given resource by the tenant ID.
+     *
+     * @param tenantId         The ID of the tenant.
+     * @param resourceTypeName The name of the {@link ResourceType}.
+     * @param resourceName     The name of the {@link ResourceType}.
+     * @return {@link Resource}
+     * @throws ConfigurationManagementException Configuration management exception.
+     */
+    Resource getResourceByTenantId(int tenantId, String resourceTypeName, String resourceName) throws
```

Comment:
this should be a default method, since we are adding new service to an existing interface

Reply:
fixed with https://github.com/wso2/carbon-identity-framework/pull/5015